### ISSUE TITLE
gmao_obs2ioda.py is never tested in ioda-converters

### ISSUE THREAD

**Issue by rhoneyager** [JCSDA-internal/ioda-converters#829]
See title. This was noticed in our code sprint. The script should be tested somewhere, but it isn't. If it is dead code, then it should be removed.

**Comment by HamidehGMAO** [comment]
Hmm, I believe @aerorahul did this work to convert the gmao's ocean obs to ioda format, But I have no where I can find this datasets. Rahul, Do you have any suggestion how to test this? should we have one test for each data (sss, sst,..)? and how I can access this data?

**Comment by aerorahul** [comment]
@HamidehGMAO @travissluka @guillaumevernieres 

I had contributed this script for converting ocean and ice observations used in the GEOS S2S system.
These observations were used for cycling SOCA with MOM6 using these set of observations.  A logical next step would have been to compare analysis statistics between SOCA MOM6 system and GEOS S2S system. 
This would have highlighted the differences due to changes in the model and analysis system.

If there is a need for adding a test for keeping this converter, a sample of input observations from GMAO can be generated using the following scripts:
1. [getGMAOOceanObs.sh](https://github.com/aerorahul/mom6-tools/blob/develop/scripts/getGMAOOceanObs.sh)
2. [GMAOobs2ioda.sh](https://github.com/aerorahul/mom6-tools/blob/develop/scripts/GMAOobs2ioda.sh)

The first script obtains the GMAO observations and constructs hourly bins that are ingested into the UMD LETKF employed at GMAO.  Hourly bins were necessary because the underlying script assigns the same datetime for all observations in the window.  This is undesirable for FGAT.
The underlying script comes from a GEOS tag `yuri-S2S-2_1_UNSTABLE` kindly provided by @yvikhlya 

The second script converts the hourly binned observations into 24 hour IODA netCDF files.  

I no longer have access to NASA Discover, though I am happy to answer any questions related to the 2 scripts.

**Comment by HamidehGMAO** [comment]
@aerorahul Thanks a lot for the information. I will take a look and let you know if I have any questions. Thanks

**Comment by aerorahul** [comment]
@HamidehGMAO 

[getGMAOOceanObs.sh](https://github.com/aerorahul/mom6-tools/blob/develop/scripts/getGMAOOceanObs.sh) calls  [ocean_obs.py](https://github.com/GEOS-ESM/UMD_Etc/blob/main/UMD_utils/ocean_obs.py)

I think one can skip the GEOS tag above and directly clone the [UMD_Etc](https://github.com/GEOS-ESM/UMD_Etc) repository and use it.  It doesn't look like it has been updated for a while.

The path to the observations are hardwired in [ocean_obs.py](https://github.com/GEOS-ESM/UMD_Etc/blob/main/UMD_utils/ocean_obs.py), so this script must be run on Discover. 

![Screen Shot 2022-03-09 at 3 26 14 PM](https://user-images.githubusercontent.com/11394126/157529921-6c698950-bee8-4544-9a2b-f0f6590934b5.png)

**Comment by HamidehGMAO** [comment]
Right! Thanks for the clarification. I found some obs on discover, I'll grab a couple of sample files and create a test for them. 
Thanks!

**Comment by travissluka** [comment]
@HamidehGMAO have you had a chance to create a test for these obs? @gthompsnJCSDA is bothering me about this.

**Comment by HamidehGMAO** [comment]
@travissluka I was able to create the obs files, but I had some issues while running the test. I'll take another look.

**Comment by gthompsnJCSDA** [comment]
> @travissluka I was able to create the obs files, but I had some issues while running the test. I'll take another look.

If you want, then I can handle this as part of our sprint for ```feature/sprint-ioda-converters``` and update the script to handle the new Data Conventions and we can wait to implement the test until this branch is ready for merging to develop.  I just need to get an obs file on Orion and I can take this on myself.

**Comment by HamidehGMAO** [comment]
@gthompsnJCSDA Thanks for your offer. I go through this today, if I couldn't figure it out. I send you a sample of obs file to take over this. Thanks!

**Comment by HamidehGMAO** [comment]
@gthompsnJCSDA  Thanks for the clarification. I didn't know you have been working on this. Please find attached 2 samples of gmao-ocean obs files. Please let me know if you have any questions. https://drive.google.com/drive/folders/1N4tkuA4c981QOsSmMP2N3NE4j128T88j?usp=sharing

**Comment by gthompsnJCSDA** [comment]
If it matters at all, the 20180415 dataset contains these 3 variables:
```
 waterTemperature
 salinity
 absoluteDynamicTopography
```

and the 20120101 file has those 3 variables plus ```seaIceFraction``` but neither sample contains other variables that the converter is hopefully capable of using, which are
```
 seaSurfaceTemperature
 iceThickness
```

**Comment by HamidehGMAO** [comment]
Yeah! It's a very sparse data set. It's hard to see if there is a date that we have all these measurements, and the [ocean_obs.py](https://github.com/GEOS-ESM/UMD_Etc/blob/main/UMD_utils/ocean_obs.py) is outdated. some of the obs directories are not valid any more. for example there is no such directory as /gpfsm/dnb42/projects/p17/gvernier/SAND_BOXES/sea_ice_da/ICE_BRIDGE/ in the ocean_obs.py any more. So, I'm guessing some of the ice data are missing here.

**Comment by travissluka** [comment]
closed with PR#899

**Comment by travissluka** [comment]
closed with PR#899
